### DISCUSSION TITLE
Clarification on game files that can be deleted

### DISCUSSION THREAD

**Discussion by connorsullivan** [pg9182/northstar-dedicated#7] [Support]
Per the README, the 3rd bullet point on "Reducing the size":

delete r2/paks/ except for {patch_master,{common,common_mp,highlight,particle_scripts}{,\(*}}.rpak ui{,\(*}.dll

I'm struggling to find a command that can actually parse that expression. Can someone please provide a command that will delete the files mentioned here?

**Comment by connorsullivan** [comment]
Additionally, it might be helpful to add a single command to the README that would delete all un-needed game files at once.

**Reply by hhollenstain** [reply]
With what @connorsullivan noted, I am also struggling how to use the string to parse file names noted. I can figuring with a simple find and or grep structure but probably helpful to put in a command or additional information to make this easier to understand.  If that isn't for parsing via command can add awk/find/grep examples to parse and purge files for others.

**Comment by pg9182** [comment]
Oh, that's a bash expansion:

```bash
$ echo {patch_master,{common,common_mp,highlight,particle_scripts}{,\(*}}.rpak ui{,\(*}.dll
patch_master.rpak common.rpak common(*.rpak common_mp.rpak common_mp(*.rpak highlight.rpak highlight(*.rpak particle_scripts.rpak particle_scripts(*.rpak ui.dll ui(*.dll
```

It's definitely confusing if you aren't used to them (and still a bit hard to read if you are), so I'll do something about that later this week.

**Reply by ReubenBTalbott** [reply]
Any update on this?

**Comment by OctoNezd** [comment]
I made a powershell script for copying all server-only files:
```
$whitelist = (Get-Content file_list.txt) -split "`n"
$whitelist | ForEach-Object {
    $filePath = "./Titanfall2/" + $_
    $targetPath = "./NSServer/" + $_
    New-Item -Force $targetPath
    Copy-Item $filePath $targetPath -Force -Recurse
}
```
[file_list.txt](https://github.com/pg9182/northstar-dedicated/files/7943604/file_list.txt)

I couldn't actually test whether script does make proper server install, cause I had a sudden realization that my VPS has two cores only so it gets stuck on "Launching the game..."
